src/lib.rs

```diff
@@ -4,32 +4,40 @@
 /// type.
 #[macro_export]
 macro_rules! serde_visitor {
-    ($name:ident, $visitor:ident) => (
+    ($name:ident, $visitor:ident, $($variant:ident),+) => (
+        #[allow(non_upper_case_globals)]
+        const $name: &'static [&'static str] = &[$(stringify!($variant)),+];
+
```

Comment:
I was actually surprised this worked! I didn't know you can have a constant and an enum/struct share the same name. I was thinking it would be better to have some internal variable name, but e.g. `$name_se__` but that isn't possible currently with macros.

I think it would be better to have this `const` defined inside the `visit_str` fn so the constant doesn't escape the into the global scope since it's an implementation detail.